spec/models/budget_spec.rb

```diff
@@ -1,5 +1,35 @@
 require 'rails_helper'
 
 RSpec.describe Budget, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  it 'has a valid factory' do
+    expect(FactoryGirl.build(:budget)).to be_valid
+  end
+
+  it 'is invalid without category' do
+    budget = FactoryGirl.build(:debt, category: nil)
+    expect(budget).to_not be_valid
+    expect(budget.errors).to have_key(:category)
+  end
+
+  it 'is invalid without an amount' do
+    budget = FactoryGirl.build(:budget, amount: nil)
+    expect(budget).to_not be_valid
+    expect(budget.errors).to have_key(:amount)
+  end
+  it 'is invalid without budget month' do
+    budget = FactoryGirl.build(:budget, budget_month: nil)
+    expect(budget).to_not be_valid
+    expect(budget.errors).to have_key(:budget_month)
+  end
+
+  it 'has many spendings' do
+    budget = FactoryGirl.create(:user_with_spendings)
+    expect(budget.spendings.length).to eq(1)
+  end
+
+  it 'is unique per budget' do
```

Comment:
doesn't really make sense for a budget to be "unique per budget". we need a better description